File: src/languages/java/binary/converter/converter.java
package binary.converter;

import binary.config.config;

public class converter {
    
    public static String BoolToStr(boolean a) {

        String o = "0";
        if (a == true) { o = "1"; }
    
        return o;
    
    }

    public static boolean StrToBool(String a) {

        boolean o = false;
        if (a.contains("1")) { o = true; }
    
        return o;
    
    }
    
    public static String CompileString(boolean ... bool) {
    
        String[] ns = new String[config.BitLimit];
    
        for (int i = 0; i < bool.length; i++) {
    
            ns[i] = BoolToStr(bool[i]);
    
        }
    
        StringBuilder s = new StringBuilder();

        for (String ch : ns) {
            s.append(ch);
        }
 
        // convert in string
        String string = s.toString();

        return string;
    
    }
    
    public static boolean[] DecompileString(String a){

        boolean[] ns = new boolean[a.length()];
        for (int i=0; i<a.split("").length; i++) {
            ns[i] = StrToBool(a.split("")[i]);
        }
        return ns;
    }
    
    public static boolean[] ReverseList(boolean[] l){
        int e=l.length-1;
        boolean[] l2= new boolean[l.length];
        for (int i=0; i<l.length; i++) {
    
            l2[e]=l[i];
            e--;
        }
        return l2;
    }
    
    public static String FixBinary(String bin){
        boolean[] a = ReverseList(DecompileString(bin));
        boolean[] a2 = new boolean[config.BitLimit];
        if (config.BitLimit-a.length < 0){
            System.out.println("FATAL ERROR: Number Is Larger Than BitLimit "+String.format("%s", config.BitLimit));
        }
        int e= a.length;
        for (int i=0; i<e; i++){
            a2[i] = a[i];
        }
        if (config.BitLimit-e != 0 || config.BitLimit-e > -1){
            for (int i=0; i<(config.BitLimit-(e)); i++) {
                a2[i+(e)] = false;
            }
        }
        return CompileString(ReverseList(a2));
    }


}

File: src/languages/go/binary/converter/converter.go
package converter

import (
	"fmt"
	"minicallib/binary/config"
	"strconv"
	"strings"
)

func BoolToStr(a bool) string {

	o := "0"
	if a == true {
		o = "1"
	}

	return o

}

func StrToBool(a string) bool {

	o := false
	if strings.Contains(a, "1") {
		o = true
	}

	return o

}

func CompileString(args []bool) string {

	ns := make([]string, config.BitLimit())

	for i := 0; i < config.BitLimit(); i++ {

		ns = append(ns, BoolToStr(args[i]))

	}

	return strings.Join(ns, "")

}

func DecompileString(a string) []bool {

	ns := make([]bool, len([]rune(a)))
	for i := 0; i < len([]rune(a)); i++ {
		ns = append(ns, StrToBool(string([]rune(a)[i])))
	}
	return ns

}

func ReverseList(l []bool) []bool {
	e := len(l) - 1
	l2 := make([]bool, len(l))
	for i := 0; i < len(l); i++ {
		l2[e] = l[i]
		e--
	}
	return l2
}

func FixBinary(bin string) string {
	a := ReverseList(DecompileString(bin))
	a2 := make([]bool, config.BitLimit())
	fmt.Println(len(DecompileString(bin)))
	if config.BitLimit()-len(a) < 0 {
		fmt.Println("FATAL ERROR: Number Is Larger Than BitLimit " + strconv.Itoa(config.BitLimit()))
	}
	e := len(a)
	for i := 0; i < e; i++ {
		a2[i] = a[i]
	}
	if config.BitLimit()-e != 0 || config.BitLimit()-e > -1 {
		for i := 0; i < (config.BitLimit() - (e)); i++ {
			a2[i+(e)] = false
		}
	}
	return CompileString(ReverseList(a2))
}
